conda_build/metadata.py

```diff
@@ -1557,9 +1557,16 @@ def get_hash_contents(self):
 
         # if dependencies are only 'target_platform' then ignore that.
         if dependencies == ["target_platform"]:
-            return {}
+            hash_contents = {}
         else:
-            return {key: self.config.variant[key] for key in dependencies}
+            hash_contents = {key: self.config.variant[key] for key in dependencies}
+
+        # include virtual packages in run
+        run_reqs = self.meta.get("requirements", {}).get("run", [])
+        virtual_pkgs = [req for req in run_reqs if req.startswith("__")]
+
+        hash_contents.update({pkg.split(" ")[0]: pkg for pkg in virtual_pkgs})
```

Comment:
Why do you split the pkg here?

Reply:
Would this help:
```suggestion
        # add name -> match spec mapping for virtual packages
        hash_contents.update({pkg.split(" ")[0]: pkg for pkg in virtual_pkgs})
```

Reply:
For context, the conda-build specs are `str` objects with up to three fields: 

```yaml
requirements:
  run:
   - pip
   - python 3.9
   - python_abi * *_cpython
```